client/ClientNetworking.cpp

```diff
@@ -421,8 +429,10 @@ bool ClientNetworking::Impl::ConnectToServer(
         ErrorLogger(network) << "ConnectToServer() : unable to connect to server at "
                              << ip_address << " due to exception: " << e.what();
     }
-    if (IsConnected())
+    if (IsConnected()) {
         m_destination = ip_address;
+        m_localhost_server = m_destination == "localhost" || m_destination == "127.0.0.1";
```

Comment:
I'm sure we can use `m_socket->remote_endpoint().address().is_loopback()`

Reply:
Change found at https://github.com/freeorion/freeorion/pull/3854/commits/e937091c4540e765549e9147757117dfbe3f4bcf#diff-4f06b6aa7be808a707f9ce9f096345a34f4cfe138aa1d1605ee9961d7611475dL481 after the removal of the indicated line (https://github.com/freeorion/freeorion/pull/3854/commits/e937091c4540e765549e9147757117dfbe3f4bcf#diff-4f06b6aa7be808a707f9ce9f096345a34f4cfe138aa1d1605ee9961d7611475dL434.)